app/controllers/saml_idp_controller.rb

```diff
@@ -38,6 +36,20 @@ def logout
 
   private
 
+  def process_fully_authenticated_user
+    link_identity_from_session_data
+
+    needs_idv = identity_needs_verification?
+    analytics.track_event(Analytics::SAML_AUTH, @result.to_h.merge(idv: needs_idv))
+
+    yield needs_idv
```

Comment:
Because this method is meant to be called with a block so that the `auth` method can `return` if the user `needs_idv`.